### PR TITLE
PUBDEV-8639 improve Uplift DRF documentation

### PR DESCRIPTION
**Pull request by maurever**
JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8639

Suggestions by @meganjkurka: 

- [x] move the parameters specific to uplift (like treatment column) to higher in the list
- [x] describe the process to calculate uplift 
- [x] add information about the uplift prediction

### CODE REVIEW

File: h2o-docs/src/product/data-science/upliftdrf.rst [hunk 1]
```diff
@@ -35,6 +36,40 @@ In a tree node the result value for a split is sum :math:`metric(P, Q) + metric(
...
+
+:math:`uplift_score_cl = TP_cl - CP_cl`
+
+Higher uplift score means more observations from treatment group respond to the offer than from control group. Which means offered treatment has positive effect. The uplift score can be negative, if more observations from control group respond to the offer without treatment.
+
+Final prediction is calculated in the same way as in DRF algorithm. Predictions for each observation is collect from all trees from an ensemble and mean prediction is returned. 
+
+When the ``predict`` method is called on test data the result frame has these columns:
+
+- ``uplift_predict``: result uplift prediction score, which is calculated as :math:`p_y1_ct1 - p_y1_ct0`
```
Comment (meganjkurka) [inline h2o-docs/src/product/data-science/upliftdrf.rst:69]: When I run uplift, I do not see this output:

- ``uplift_predict``: result uplift prediction score, which is calculated as :math:`p_y1_ct1 - p_y1_ct0`
- ``p_y1_ct1``: probability the response is 1 if the row is from the treatment group
- ``p_y1_ct0``: probability the response is 1 if the row is from the control group
- 
I see this:
Screen Shot 2022-04-05 at 6 08 23 AM

**Review by meganjkurka — COMMENTED** [review]
I don't see the same predict output as what's documented

File: h2o-docs/src/product/data-science/upliftdrf.rst (hunk 1 above)
Comment (maurever) [reply h2o-docs/src/product/data-science/upliftdrf.rst:69]: Thanks, @meganjkurka, for reporting this issue; the bug is fixed here: https://github.com/h2oai/h2o-3/pull/6138. So then, this documentation is correct.

File: h2o-docs/src/product/data-science/upliftdrf.rst [hunk 2]
```diff
@@ -35,6 +36,40 @@ In a tree node the result value for a split is sum :math:`metric(P, Q) + metric(
 
 You can read more information about ``uplift_metric`` on parameter specification page: `uplift_metric <algo-params/uplift_metric.html>`__.
 
+Uplift tree and prediction
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tree-based algorithm means in every tree, it takes information about treatment/control group assignment and information about response directly into a decision about splitting a node. So there is only one tree for every class and both groups - no separate trees for treatment groups data and control groups data.
```
Comment (valenad1) [inline h2o-docs/src/product/data-science/upliftdrf.rst:42]: I would say "**Uplift** Tree-based algorithm means in every tree,...."

**Review by valenad1 — APPROVED** [review]
Looks good to me, thanks 👍

**Comment by valenad1** [comment]
If it goes to the major release, I would merge it after https://github.com/h2oai/h2o-3/pull/6138 to have it consistent.

**Comment by maurever** [comment]
@meganjkurka, are you ok with the improvements? :)

**Pull request merged**